### DISCUSSION TITLE
Microsoft.Data.SqlClient is not supported on this platform.

### DISCUSSION THREAD

**Discussion by omerrasyonet** [Excel-DNA/ExcelDna#497] [Q&A] [answered]
Hi,

I'm using Microsoft.Data.SqlClient in a net6.0-windows ExcelDna project and I received this message : "Microsoft.Data.SqlClient is not supported on this platform." How can I fix this problem

Thank you

![PlatformNotSupported](https://user-images.githubusercontent.com/80813206/185369831-dc6dc649-cc6f-4657-8c4b-f68717e7581f.png)

**Comment by govert** [comment]
I think this is similar to the discussion here: https://groups.google.com/g/exceldna/c/4PQLjPj76N4
Maybe you can copy the files from the platform-specific directory into the add-in directory.

**Comment by govert** [comment, marked as answer]
In particular, it looks like you'll need to copy the Microsoft.Data.SqlClient.dll from the runtimes\win\lib\netcoreapp3.1\ and also Microsoft.Data.SqlClient.SNI.dll from runtimes\win-x64\native\. The first of these is overwritten with every build, so you need to re-copy every time, or add a post-build task or similar.
Then at least I get past the PlatformNotSupportedException. I don't have an easy setup to test further than this.

**Reply by omerrasyonet** [reply]
it works, thank you all
